includes/multi-currency/MultiCurrency.php

```diff
@@ -589,6 +595,42 @@ public function recalculate_cart() {
 		WC()->cart->calculate_totals();
 	}
 
+	/**
+	 * Displays a notice on the frontend informing the customer of the
+	 * automatic currency switch.
+	 */
+	public function display_geolocation_currency_update_notice() {
+		$current_currency = $this->get_selected_currency();
+		$store_currency   = get_option( 'woocommerce_currency' );
+		$country          = $this->locale->get_country_by_customer_location();
+		$currencies       = get_woocommerce_currencies();
+
+		// Do not display notice if using the store's default currency.
+		if ( $store_currency === $current_currency->get_code() ) {
+			return;
+		}
+
+		$message = sprintf(
+			/* translators: %1 User's country, %2 Selected currency name, %3 Default store currency name */
+			__( 'We noticed you\'re visiting from %1$s. We\'ve updated our prices to %2$s for your shopping convenience. Use <a>%3$s</a> instead.', 'woocommerce-payments' ),
```

Comment:
The link should be the whole `Use XXXXXXX instead` sentence, instead of just the name of the currency.

```suggestion
			__( 'We noticed you\'re visiting from %1$s. We\'ve updated our prices to %2$s for your shopping convenience. <a>Use %3$s instead.</a>', 'woocommerce-payments' ),
```

Reply:
Good call! I've implemented that in https://github.com/Automattic/woocommerce-payments/pull/2506/commits/ee3c790e5fb12a90b950423492cad17cc9bf75ae.